lnrpc/wtclientrpc/wtclient.go

```diff
@@ -0,0 +1,298 @@
+// WatchtowerClientWatchtowerClient gRPC service.
+var _ WatchtowerClientServer = (*WatchtowerClient)(nil)
+
+// New returns a new instance of the wtclientrpc WatchtowerClient sub-server.
+// We also return the set of permissions for the macaroons that we may create
+// within this method. If the macaroons we need aren't found in the filepath,
+// then we'll create them on start up. If we're unable to locate, or create the
+// macaroons we need, then we'll return with an error.
+func New(cfg *Config) (*WatchtowerClient, lnrpc.MacaroonPerms, error) {
+	return &WatchtowerClient{*cfg}, macPermissions, nil
+}
+
+// Start launches any helper goroutines required for the WatchtowerClient to
+// function.
+//
+// NOTE: This is part of the lnrpc.SubWatchtowerClient interface.
+func (c *WatchtowerClient) Start() error {
+	return nil
+}
+
+// Stop signals any active goroutines for a graceful closure.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (c *WatchtowerClient) Stop() error {
+	return nil
+}
+
+// Name returns a unique string representation of the sub-server. This can be
+// used to identify the sub-server and also de-duplicate them.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (c *WatchtowerClient) Name() string {
+	return subServerName
+}
+
+// RegisterWithRootServer will be called by the root gRPC server to direct a sub
+// RPC server to register itself with the main gRPC root server. Until this is
+// called, each sub-server won't be able to have requests routed towards it.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (c *WatchtowerClient) RegisterWithRootServer(grpcServer *grpc.Server) error {
+	// We make sure that we register it with the main gRPC server to ensure
+	// all our methods are routed properly.
+	RegisterWatchtowerClientServer(grpcServer, c)
+
+	log.Debugf("WatchtowerClient RPC server successfully registered with " +
+		"root gRPC server")
+
+	return nil
+}
+
+// isActive returns nil if the watchtower client is initialized so that we can
+// process RPC requests.
+func (c *WatchtowerClient) isActive() error {
+	if c.cfg.Active {
+		return nil
+	}
+	return ErrWtclientNotActive
+}
+
+// AddTower adds a new watchtower reachable at the given address and considers
+// it for new sessions. If the watchtower already exists, then any new addresses
+// included will be considered when dialing it for session negotiations and
+// backups.
+func (c *WatchtowerClient) AddTower(ctx context.Context,
+	req *AddTowerRequest) (*AddTowerResponse, error) {
+
+	if err := c.isActive(); err != nil {
+		return nil, err
+	}
+
+	pubKey, err := btcec.ParsePubKey(req.Pubkey, btcec.S256())
+	if err != nil {
+		return nil, err
+	}
+	addr, err := lncfg.ParseAddressString(
+		req.Address, strconv.Itoa(watchtower.DefaultPeerPort),
+		c.cfg.Resolver,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("invalid address %v: %v", req.Address, err)
+	}
+	towerAddr := &lnwire.NetAddress{
```

Comment:
Style nit: could use a space between the `towerAddr` creation and the set up above.